crates/uv-client/src/cached_client.rs

```diff
@@ -519,6 +565,133 @@ impl CachedClient {
         };
         Ok((response, cache_policy))
     }
+
+    /// Perform a [`CachedClient::get_serde`] request with a default retry strategy.
+    #[instrument(skip_all)]
+    pub async fn get_serde_with_retry<
+        Payload: Serialize + DeserializeOwned + 'static,
+        CallBackError: std::error::Error + 'static,
+        Callback,
+        CallbackReturn,
+    >(
+        &self,
+        req: Request,
+        cache_entry: &CacheEntry,
+        cache_control: CacheControl,
+        response_callback: Callback,
+    ) -> Result<Payload, CachedClientError<CallBackError>>
+    where
+        Callback: Fn(Response) -> CallbackReturn,
+        CallbackReturn: Future<Output = Result<Payload, CallBackError>>,
+    {
+        let payload = self
+            .get_cacheable_with_retry(req, cache_entry, cache_control, |resp| async {
+                let payload = response_callback(resp).await?;
+                Ok(SerdeCacheable { inner: payload })
+            })
+            .await?;
+        Ok(payload)
+    }
+
+    /// Perform a [`CachedClient::get_cacheable`] request with a default retry strategy.
+    ///
+    /// See: <https://github.com/TrueLayer/reqwest-middleware/blob/8a494c165734e24c62823714843e1c9347027e8a/reqwest-retry/src/middleware.rs#L137>
+    #[instrument(skip_all)]
+    pub async fn get_cacheable_with_retry<
+        Payload: Cacheable,
+        CallBackError: std::error::Error + 'static,
+        Callback,
+        CallbackReturn,
+    >(
+        &self,
+        req: Request,
+        cache_entry: &CacheEntry,
+        cache_control: CacheControl,
+        response_callback: Callback,
+    ) -> Result<Payload::Target, CachedClientError<CallBackError>>
+    where
+        Callback: Fn(Response) -> CallbackReturn,
+        CallbackReturn: Future<Output = Result<Payload, CallBackError>>,
+    {
+        let mut n_past_retries = 0;
+        let start_time = SystemTime::now();
```

Comment:
Let's use a monotonic time source here instead.

(maybe that could be my first patch to `uv` :)).

Reply:
I can't comment on system clocks (@BurntSushi probably knows), but if we switch to `Instant`, we should upstream that to https://github.com/TrueLayer/reqwest-middleware/blob/8a494c165734e24c62823714843e1c9347027e8a/reqwest-retry/src/middleware.rs#L144

Reply:
Yes, both of these should 100% be monotonic time, or `std::time::Instant`.

For upstream, it's not just `reqwest-retry`, but in the `retry_policies` crate: https://docs.rs/retry-policies/latest/retry_policies/trait.RetryPolicy.html#tymethod.should_retry